### ISSUE TITLE
The page com.auth0.lock.LockActivity crashes after minimizing the app

### ISSUE THREAD

**Issue by Kolyall** [auth0/Lock.Android#164]
The page com.auth0.lock.`LockActivity` crashes with this log.
Steps to reproduce:

1. start `LockActivity` with this code:
  ```
  Intent lockIntent = new Intent(this, LockActivity.class);
  startActivity(lockIntent);
  ```
2. Minimize your app by clicking on Home button on device.
3. In Android Studio click on "Terminate Application"
4. Open your application on device again.
BUG:
 App crashes on library.

LOG:

```
E/AndroidRuntime: FATAL EXCEPTION: main
Process: com.github.pires.obd.reader, PID: 10630
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.github.pires.obd.reader/com.auth0.lock.LockActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'com.auth0.core.Connection com.auth0.lock.Configuration.getDefaultDatabaseConnection()' on a null object reference
   at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3124)
   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3233)
   at android.app.ActivityThread.access$1000(ActivityThread.java:197)
   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656)
   at android.os.Handler.dispatchMessage(Handler.java:102)
   at android.os.Looper.loop(Looper.java:145)
   at android.app.ActivityThread.main(ActivityThread.java:6856)
   at java.lang.reflect.Method.invoke(Native Method)
   at java.lang.reflect.Method.invoke(Method.java:372)
   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'com.auth0.core.Connection com.auth0.lock.Configuration.getDefaultDatabaseConnection()' on a null object reference
   at com.auth0.lock.fragment.DatabaseLoginFragment.onCreate(DatabaseLoginFragment.java:102)
   at com.auth0.lock.fragment.SocialDBFragment.onCreate(SocialDBFragment.java:42)
   at android.support.v4.app.Fragment.performCreate(Fragment.java:1939)
   at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1029)
   at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1248)
   at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1230)
   at android.support.v4.app.FragmentManagerImpl.dispatchCreate(FragmentManager.java:2037)
   at android.support.v4.app.FragmentController.dispatchCreate(FragmentController.java:154)
   at android.support.v4.app.FragmentActivity.onCreate(FragmentActivity.java:289)
   at com.auth0.lock.LockActivity.onCreate(LockActivity.java:89)
   at android.app.Activity.performCreate(Activity.java:6550)
   at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1120)
   at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3077)
   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3233)
   at android.app.ActivityThread.access$1000(ActivityThread.java:197)
   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656)
   at android.os.Handler.dispatchMessage(Handler.java:102)
   at android.os.Looper.loop(Looper.java:145)
   at android.app.ActivityThread.main(ActivityThread.java:6856)
   at java.lang.reflect.Method.invoke(Native Method)
   at java.lang.reflect.Method.invoke(Method.java:372)
   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199) 
```

**Comment by nikolaseu** [comment]
Hi, I can reproduce the crash.
I guess it's not a common use case however, I'll see what we could do and comment back later.
Thanks for reporting

**Comment by vpetkar** [comment]
I was also affected by this issue. Thought it was a problem with my code, however an uninstall and install seems to have solved the issue.

**Comment by nicolas9019** [comment]
I'm also getting the similar Issue with the lock activity while trying to launch the auth0 login page. Past 2 days wasted trying to figure out what wrong. I'm getting this issue on the latest android studio on windows.

**Comment by jayhuang75** [comment]
Any update on this?

**Comment by hzalaz** [comment]
Closed by #318
